src/main/java/net/smartcosmos/dao/objects/impl/ObjectPersistenceService.java

```diff
@@ -133,11 +129,11 @@ public ObjectResponse create(String accountUrn, ObjectCreate createObject) {
         for (ObjectEntity singleResult : queryResult)
         {
             // created is set at object creation time, and lastModified is not
-            Date singleResultLastModified = singleResult.getLastModified();
+            Long singleResultLastModified = singleResult.getLastModified();
             if (singleResultLastModified == null){
                 singleResultLastModified = singleResult.getCreated();
             }
-            if(!modifiedAfterDateSpecified || singleResultLastModified.after(modifiedAfterDate))
```

Comment:
We cannot be doing this.  We need to make the database do the work by invoking the query with a comparison for the `lastUpdatedTimestamp`.  We should never retrieve results and iterate or inspect the return wen that can be most effectively done by the database.